src/__tests__/node_redis.spec.js

```diff
@@ -118,13 +119,75 @@ tape('NodeJS Redis', function (t) {
         await client.destroy();
 
         // close server connection
         server.close().then(assert.end);
       });
   });
 
+  t.test('Connection ready and timed out', assert => {
+    const readyTimeout = 0.1; // 100 millis
+    const configWithShortTimeout = { ...config, startup: { readyTimeout } };
+    const sdk = SplitFactory(configWithShortTimeout);
+    const client = sdk.client();
+
+    const start = Date.now();
+    let readyTimestamp;
+
+    client.getTreatment('UT_Segment_member', 'always-on').then(treatment => {
+      assert.equal(treatment, 'on', 'Evaluations using Redis storage should be correct and resolved once Redis connection is stablished');
+    });
+    client.track('nicolas@split.io', 'user', 'test.redis.event', 18).then(result => {
+      assert.true(result, 'If the event was succesfully queued the promise will resolve to true once Redis connection is stablished');
+    });
+
+    // SDK_READY_TIMED_OUT event must be emitted after 100 millis
+    client.on(client.Event.SDK_READY_TIMED_OUT, () => {
+      const delay = Date.now() - start;
+      assert.true(nearlyEqual(delay, readyTimeout * 1000), 'SDK_READY_TIMED_OUT event must be emitted after 100 millis');
+    });
+
+    // alse, ready promise must be rejected after 100 millis
+    client.ready().catch(() => {
+      const delay = Date.now() - start;
+      assert.true(nearlyEqual(delay, readyTimeout * 1000), 'Ready promise must be rejected after 100 millis');
+
+      // initialize server to emit SDK_READY
+      initializeRedisServer().then(async (server) => {
+        readyTimestamp = Date.now();
+
+        try {
+          await client.ready();
+          assert.fail('Ready promise keeps being rejected until SDK_READY is emitted');
+        } catch (error) {
+          assert.pass('Ready promise keeps being rejected until SDK_READY is emitted');
+        }
+
+        client.on(client.Event.SDK_READY, async () => {
+          const delay = Date.now() - readyTimestamp;
+          assert.true(nearlyEqual(delay, 0, 100), 'SDK_READY event must be emitted soon once Redis server is connected');
+
+          await client.ready();
+          assert.pass('Ready promise is resolved once SDK_READY is emitted');
+
+          // some asserts to test regular usage
+          assert.equal(await client.getTreatment('UT_Segment_member', 'UT_IN_SEGMENT'), 'on', 'Evaluations using Redis storage should be correct.');
+          assert.equal(await client.getTreatment('other', 'UT_IN_SEGMENT'), 'off', 'Evaluations using Redis storage should be correct.');
+          assert.true(await client.track('nicolas@split.io', 'user', 'test.redis.event', 18), 'If the event was succesfully queued the promise will resolve to true');
+          assert.false(await client.track(), 'If the event was NOT succesfully queued the promise will resolve to false');
+
+          await client.destroy();
+
+          // close server connection
+          server.close().then(assert.end);
```

Comment:
This doesn't seem to actually wait for the statements on: https://github.com/splitio/javascript-client/pull/422/files#diff-61efb51f5e5ba9bdba3f39bd1fb3fcdcR139-R144
(the two assertions for the SDK yielding the right result once the individual promises resolve)

Should we add meassurements to ensure that those assertions run? using plan or keeping track with a promise all between those and the server.close() ?